### PR TITLE
replace Travis with CircleCI

### LINKED ISSUE #80: replace TravisCI with CircleCI
We use CircleCI everywhere else, so it'd be nice to not have to worry about the differences between Circle and Travis

Thoughts @segmentio/gateway?

### PR DESCRIPTION
**Pull request by stephenmathieson**
This patch replaces Travis with CircleCI. This prevents the context-switching between Travis and Circle (since we use Circle for nearly everything else), and gives us more flexibility for handling builds.

This patch also introduces automatically published builds (to npm) and automatically generated `analytics-node.js` versions.

Closes #80.

@segmentio/gateway 

----

NOTE - Travis build is failing because I removed the `.travis.yml` file. Once approved, I'll disable Travis and rebuild.

### CODE REVIEW

File: RELEASING.md [hunk 1]
```diff
@@ -1,4 +1,13 @@
...
+
+```
+$ git changelog --tag <version>
+$ vim package.json # update ".version"
+$ git release <version>
+```
+
+Where `<version>` represents a valid [semver](http://semver.org) version number.
+
+CircleCI will handle rebuilding `analytics-node.js` and publishing to npm for you.sion>`.
```
Comment (timhudson) [inline RELEASING.md]: `you.sion>`?

Comment (stephenmathieson) [inline RELEASING.md]: lol that moment when you realize how bad at using `vim` you are

File: circle.yml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+test:
+  override:
+    - nvm use 0.12; npm rebuild > /dev/null; make test
+    - nvm use 4.1; npm rebuild > /dev/null; make test
+    - nvm use 4; npm rebuild > /dev/null; make test
+    - nvm use 6; npm rebuild > /dev/null; make test
+
+deployment:
+  publish:
+    tag: /v?[0-9]+(\.[0-9]+)*(-.+)?/
```
Comment (timhudson) [inline circle.yml]: This repo leaves out the leading `v` for tags.

Comment (stephenmathieson) [inline circle.yml]: Yeah, this is just our "standard" regex. I'll update tho.

File: circle.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+    - nvm use 4; npm rebuild > /dev/null; make test
+    - nvm use 6; npm rebuild > /dev/null; make test
+
+deployment:
+  publish:
+    tag: /[0-9]+\.[0-9]+\.[0-9]+/
+    commands:
+      - make analytics-node.js
+      - git add analytics-node.js
+      - git commit -m "analytics-node.js@$CIRCLE_TAG [ci skip]"
```
Comment (timhudson) [inline circle.yml]: 👍 good addition

Comment (stephenmathieson) [inline circle.yml]: Yeah, I've cut a few releases of this thing and always forgot to rebuild this file. Figured I should make CI do it so I don't have to remember :P

**Pull request merged**